resolvers/README.md

```diff
@@ -204,6 +225,14 @@ the absolute path to the file making the import (`/some/path/to/module.js`)
 an object provided via the `import/resolver` setting. `my-cool-resolver` will get `["some", "stuff"]` as its `config`, while
 `node` will get `{ "paths": ["a", "b", "c"] }` provided as `config`.
 
+##### `undefined`
+
+For compatibility reason, the 4th argument is always passed as `undefined`. Take [TypeScript resolver](https://github.com/import-js/eslint-import-resolver-typescript/blob/c45039e5c310479c1e178c2180e054380facbadd/src/index.ts#L69) for example. It uses the 4th argument to pass the optional [unrs-resolver] `ResolverFactory` instance to support both `v2` and `v3` interface at the same time.
```

Comment:
TBH, this `undefined` feels like bad API design. If we need to add a bandit to the previous design failure, I'd rather introduce a breaking change. Alternatively, we can use some kind of [JavaScript context implementation](https://blog.skk.moe/post/context-in-javascript/) to do this, something like `useRuleContext()` and `useResolvedTsConfig()`.

Reply:
The goal is 100% compatibility at least in this PR, v2 interface has been frozen for a long time, I don't know how a breaking change would work even in a major version.

This is not a design, just how it has to be.

Reply:
How are they different with the lazy getter? I don't want over-enginering here.

Reply:
> The goal is 100% compatibility at least in this PR, v2 interface has been frozen for a long time, I don't know how a breaking change would work even in a major version.
> 
> This is not a design, just how it has to be.

What I am proposing here is that we implement a context that lives outside of the full resolve closure scope. The resolver would be doing this instead:

```ts
import { useRuleContext, useResolvedTsConfig } from 'eslint-plugin-import-x/resolver-utils';

// Heck, even the old v2 interface can benefits!
exports.resolveImport = (stillOnly, TwoArguments) => {
  const context = useRuleContext();
  const tsconfig = useResolvedTsConfig();
};
```

Reply:
How about push a commit to this PR/branch?